app/src/main/kotlin/com/wire/android/ui/home/conversationslist/ConversationListViewModel.kt

```diff
@@ -174,94 +176,105 @@ class ConversationListViewModelImpl @AssistedInject constructor(
-    override val conversationListState: ConversationListState
-        get() = if (usePagination) {
-            ConversationListState.Paginated(
-                conversations = conversationsPaginatedFlow,
-                domain = currentAccount.domain
-            )
-        } else {
-            notPaginatedConversationListState
+    override var conversationListState by mutableStateOf(
+        when (usePagination) {
+            true -> ConversationListState.Paginated(conversations = conversationsPaginatedFlow, domain = currentAccount.domain)
+            false -> ConversationListState.NotPaginated()
         }
+    )
+        private set
 
     init {
+        observeSelfUserLegalHoldState()
         if (!usePagination) {
-            viewModelScope.launch {
-                searchQueryFlow
-                    .debounce { if (it.isEmpty()) 0L else DEFAULT_SEARCH_QUERY_DEBOUNCE }
-                    .onStart { emit("") }
-                    .distinctUntilChanged()
-                    .flatMapLatest { searchQuery: String ->
-                        observeConversationListDetailsWithEvents(
-                            fromArchive = conversationsSource == ConversationsSource.ARCHIVE,
-                            conversationFilter = conversationsSource.toFilter()
-                        ).combine(observeLegalHoldStateForSelfUser()) { conversations, selfUserLegalHoldStatus ->
-                            conversations.map { conversationDetails ->
-                                conversationDetails.toConversationItem(
-                                    userTypeMapper = userTypeMapper,
-                                    searchQuery = searchQuery,
-                                    selfUserTeamId = observeSelfUser().firstOrNull()?.teamId
-                                ).hideIndicatorForSelfUserUnderLegalHold(selfUserLegalHoldStatus)
-                            } to searchQuery
-                        }
-                    }
-                    .map { (conversationItems, searchQuery) ->
-                        if (searchQuery.isEmpty()) {
-                            conversationItems.withFolders(source = conversationsSource).toImmutableMap()
-                        } else {
-                            searchConversation(
-                                conversationDetails = conversationItems,
-                                searchQuery = searchQuery
-                            ).withFolders(source = conversationsSource).toImmutableMap()
-                        }
+            observeNonPaginatedSearchConversationList()
+        }
+    }
+
+    private fun observeSelfUserLegalHoldState() {
+        viewModelScope.launch {
+            observeLegalHoldStateForSelfUser()
+                .map { it is LegalHoldStateForSelfUser.Enabled }
+                .flowOn(dispatcher.io())
+                .collect { isSelfUserUnderLegalHoldFlow.emit(it) }
+        }
+    }
```

Comment:
Instead of having a separate `fun` with collect you can have:
```
val isSelfUserUnderLegalHoldFlow = observeLegalHoldStateForSelfUser()
    .map { it is LegalHoldStateForSelfUser.Enabled }
    .flowOn(dispatcher.io())
    .shareIn(viewModelScope, SharingStarted.WhileSubscribed(), 1)

```

Reply:
No, because `observeLegalHoldStateForSelfUser()` is actually a suspend fun for some reason so it needs to be called from a coroutine or another suspend fun 🙃 
I could use `runBlocking` but that would probably slow down the loading of the conversation list